behat-tests/features/acquia_lift_element_variations.feature

```diff
@@ -1,170 +1,198 @@
-#Assumes Marketer role.
+@api @javascript @insulated @campaign
 Feature: Personalize elements variations can be edited for an existing campaign.
   In order to manage element variations in context
   As a site marketer
   I want the ability to add, edit, and delete existing personalize element variations in context.
 
-  @api @javascript @campaign
   Scenario: Add personalize elements to a campaign
+    # I have a campaign.
+    # I login with the marketer role.
+    # I am on an article page.
     Given "acquia_lift" agents:
-      | machine_name                    | label |
+      | machine_name                    | label                           |
       | testing-campaign-add-variations | Testing campaign add variations |
-    And I am logged in as a user with the "Marketer" role
+    And I am logged in as a user with the "access administration pages,access toolbar,administer visitor actions,manage personalized content" permission
```

Comment:
Ah that's the reason why you did not go for the permission approach - I was sure there was a reason. =D

The reason I was going straight to permissions, was because I thought the step was perfect for the use. This is because the plain idea the reader needs to know is "user has permissions xxx and yyy" instead of "user has a role of aaa that in term has permissions yyy, zzz". If I had went for "Given I have the role Marketer in the system", I would have had to comment "# Marketer has permissions to www,xxx,yyy,zzz but nothing else special" before each use anyway - so my (assume actual non-tech Marketers) readers would immediately know what Marketer means without reading code.

This really boils down to the fact that we are going through all the trouble implementing and maintaining this very expensive Gherkins language is only because we are now writing an non-tech readable article instead of a program, so I feel many programmer habits would have to give in to editor habits.

When we need to update the definition of "Marketer" in the future, we can always search and replace by utilizing the "marketer" comment I have already put on top of each use.